conans/model/build_info.py

```diff
@@ -206,6 +218,12 @@ def _raise_if_mixing_components(self):
             raise ConanException("self.cpp_info.components cannot be used with self.cpp_info configs"
                                  " (release/debug/...) at the same time")
 
+        # Raise on component name
+        for comp_name in self.components:
```

Comment:
Also as a oneliner: ``if any(comp_name == name for comp_name in self.components)``